src/xdr-type.js

```diff
@@ -170,6 +170,48 @@ function decodeInput(input, format) {
   }
 }
 
+/**
+ * Provides a "duck typed" version of the native `instanceof` for read/write.
+ *
+ * "Duck typing" means if the parameter _looks like_ and _acts like_ a duck
+ * (i.e. the type we're checking), it will be treated as that type.
+ *
+ * In this case, the "type" we're looking for is "like XdrCompositeType",
+ * meaning serialization, but also conditioned on a particular subclass of
+ * "XdrCompositionType" (e.g. {@link Union} which extends XdrCompositeType ->
+ * XdrType).
+ *
+ * This makes the package resilient to downstream systems that may be combining
+ * many versions of a package across its stack that are technically compatible
+ * but fail `instanceof` checks due to cross-pollination.
+ */
+export function isSerializableIsh(value, subtype) {
+  const hasConstructor = (instance, name) => {
```

Comment:
This doesn't capture any context of the containing function. And seems useful on its own.

Reply:
Good point!